src/build.d

```diff
@@ -898,7 +897,11 @@ void processEnvironment()
     {
         dflags ~= ["-fsanitize="~env["ENABLE_SANITIZERS"]];
     }
-    flags["DFLAGS"] ~= dflags;
+
+    if (auto ptr = "DFLAGS" in flags) // Retain user-defined flags
```

Comment:
What about:
```D
flags["DFLAGS"] = flags.get("DFLAGS", "") ~ dflags;
```

Reply:
I initially wanted to use `update`  but hit an ldc bug. Changed to your approach.